odo/backends/tests/test_sql.py

```diff
@@ -278,6 +278,24 @@ def test_discover_oracle_intervals(freq):
     assert discover(t) == dshape('var * {dur: ?timedelta[unit="%s"]}' % freq)
 
 
+def test_mssql_types():
+    typ = sa.dialects.mssql.BIT()
+    t = sa.Table('t', sa.MetaData(), sa.Column('bit', typ))
+    assert discover(t) == dshape('var * {bit: ?bool}')
```

Comment:
could you change these to `datashape.util.assert_dshape_equal`. This function provides the path to any attributes that are invalid, for example:

```
In [4]: assert_dshape_equal(
    dshape('var * {a: var * var * {b: int32, c: int64}}'),
    dshape('var * {a: var * var * {b: int32, c: int32}}'),
)
---------------------------------------------------------------------------
AssertionError
  ...
AssertionError: ctype names do not match: 'int32' != 'int64'
path: _.measure['a'].measure['c'].name
```